assets/src/edit-story/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -166,9 +208,26 @@ function useMediaUploadQueue() {
 
             try {
               const attachment = await uploadFile(file, additionalData);
+              const { poster, posterId } = await uploadVideoPoster(
+                attachment.id,
+                fileName,
+                posterFile
+              );
```

Comment:
What worries me here is that if `uploadFile` works, but `uploadVideoPoster` _fails_, it will cancel the whole upload. That doesn't seem ideal.

It might be better to do this whole `uploadVideoPoster` call in a separate effect. Perhaps here:

https://github.com/google/web-stories-wp/blob/17169f3fc9829f9e741d779c184f2ac9bcabddfa/assets/src/edit-story/app/media/useUploadMedia.js#L138-L154

If it fails it shouldn't block the `updateMediaElement()` and `onUploadSuccess()` calls from running though.

Alternatively, it could be a new effect in `useMediaUploadQueue` with some new reducer, but that seems a bit overkill.

Reply:
So move this to another effect after the video is uploaded. I am worried by doing this, other part of the code that type and backfill poster images will kick in and generate a poster at the same time. I am worried that the code we try generate poster image twice again. Doing it directly after upload of video is by design.

Reply:
Of course we'd need to make sure that it doesn't clash with other backfilling logic.

An alternative would be to move the `uploadVideoPoster()` out of this `try {} catch {}` block, into a new one. Something like this:

```js

let newResource = resource;

try {
	const attachment = await uploadFile(newFile, additionalData);
	const res = getResourceFromAttachment(attachment);
	newResource = {
	  ...newResource,
	  poster: newResource.poster || resource.poster,
	};
} catch {
	cancelUploading({ id, error });
}

try {
	  const { poster, posterId } = await uploadVideoPoster(
		    attachment.id,
		    fileName,
		    posterFile
	  );

	newResource = {...}
} catch {
	// do nothing?
}

finishUploading()
```

Again, a failing `uploadVideoPoster` call should not fail the whole upload.